src/Workspaces/VisualBasic/Portable/Simplification/VisualBasicNameReducer.Rewriter.vb

```diff
@@ -14,100 +14,51 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Simplification
                 MyBase.New(optionSet, cancellationToken)
             End Sub
 
-            Public Overrides Function VisitGenericName(node As GenericNameSyntax) As SyntaxNode
+            Private Function SimplifyVisit(Of TExpression As SyntaxNode)(
```

Comment:
Same concern about lambda allocations.  Please cache the delegates in the constructor and use the cached values in the code.